src/pycares/__init__.py

```diff
@@ -528,28 +528,30 @@ def set_local_ip(self, ip):
         else:
             raise ValueError("invalid IP address")
 
-    def getnameinfo(self, ip_port, flags, callback):
+    def getnameinfo(self, address, flags, callback):
         if not callable(callback):
             raise TypeError("a callable is required")
 
-        ip, port = ip_port
-
-        if port < 0 or port > 65535:
-            raise ValueError("port must be between 0 and 65535")
-
-        sa4 = _ffi.new("struct sockaddr_in*")
-        sa6 = _ffi.new("struct sockaddr_in6*")
-
-        if _lib.ares_inet_pton(socket.AF_INET, ascii_bytes(ip), _ffi.addressof(sa4.sin_addr)) == 1:
+        if len(address) == 2:
+            (ip, port) = address
+            sa4 = _ffi.new("struct sockaddr_in*")
+            if _lib.ares_inet_pton(socket.AF_INET, ascii_bytes(ip), _ffi.addressof(sa4.sin_addr)) != 1:
+                raise ValueError("Invalid IPv4 address %r" % ip)
             sa4.sin_family = socket.AF_INET
             sa4.sin_port = socket.htons(port)
             sa = sa4
-        elif _lib.ares_inet_pton(socket.AF_INET6, ascii_bytes(ip), _ffi.addressof(sa6.sin6_addr)) == 1:
+        elif len(address) == 4:
```

Comment:
I thought I had typed this but I can no longer see it. Oh well, here it goes. Python's `socket.getnameinfo` supports passing a 2 item tuple also on IPv6. I guess it sets those to 0. Any reason not to do that here?

Reply:
No, I'm strongly confident that for IPv6 it always must be a 4-item tuple. I will fix CPython.